functions/New-DbaAgentOperator.ps1

```diff
@@ -0,0 +1,327 @@
+            # Check the start time
+            if (-not $WeekdayStartTime -and $Force) {
+                $WeekdayStartTime = '000000'
+                Write-Message -Message "Weekday Start time was not set. Force is being used. Setting it to $WeekdayStartTime" -Level Verbose
+            } elseif (-not $WeekdayStartTime) {
+                Stop-Function -Message "Please enter Weekday Start Time or use -Force to use defaults."
+                return
+            } elseif ($WeekdayStartTime -notmatch $RegexTime) {
+                Stop-Function -Message "Weekday Start time $WeekdayStartTime needs to match between '000000' and '235959'"
+                return
+            }
+
+            # Check the end time
+            if (-not $WeekdayEndTime -and $Force) {
+                $WeekdayEndTime = '235959'
+                Write-Message -Message "Weekday End time was not set. Force is being used. Setting it to $WeekdayEndTime" -Level Verbose
+            } elseif (-not $WeekdayEndTime) {
+                Stop-Function -Message "Please enter a Weekday End Time or use -Force to use defaults."
+                return
+            } elseif ($WeekdayEndTime -notmatch $RegexTime) {
+                Stop-Function -Message "Weekday End time $WeekdayEndTime needs to match between '000000' and '235959'"
+                return
+            }
+        }
+
+        if ($IsFailsafeOperator -and ($FailsafeNotificationMethod -notin ('NotifyMail', 'NotifyPager'))) {
+            Stop-Function -Message "You must specify a notifiation method for the failsafe operator."
+            return
+        }
+
+        #Format times
+        if ($SaturdayStartTime) {
+            $SaturdayStartTime = $SaturdayStartTime.Insert(4, ':').Insert(2, ':')
+        }
+        if ($SaturdayEndTime) {
+            $SaturdayEndTime = $SaturdayEndTime.Insert(4, ':').Insert(2, ':')
+        }
+
+        if ($SundayStartTime) {
+            $SundayStartTime = $SundayStartTime.Insert(4, ':').Insert(2, ':')
+        }
+        if ($SundayEndTime) {
+            $SundayEndTime = $SundayEndTime.Insert(4, ':').Insert(2, ':')
+        }
+
+        if ($WeekdayStartTime) {
+            $WeekdayStartTime = $WeekdayStartTime.Insert(4, ':').Insert(2, ':')
+        }
+        if ($WeekdayEndTime) {
+            $WeekdayEndTime = $WeekdayEndTime.Insert(4, ':').Insert(2, ':')
+        }
+
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $serverSqlCredential
+            } catch {
+                Stop-Function -Message "Failed" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            $failsafe = $server.JobServer.AlertSystem | Select-Object FailSafeOperator
+
+            if ((Get-DbaAgentOperator -SqlInstance $instance -Operator $Operator).Count -ne 0) {
```

Comment:
```suggestion
            if ((Get-DbaAgentOperator -SqlInstance $server -Operator $Operator).Count -ne 0) {
```

Should pass in the `$server` object as it will include the credential if one is provided.